packages/angular/cli/utilities/package-manager.ts

```diff
@@ -6,55 +6,87 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import { execSync } from 'child_process';
-import { existsSync } from 'fs';
+import { exec, execSync } from 'child_process';
+import { constants, promises as fs } from 'fs';
 import { join } from 'path';
 import { satisfies, valid } from 'semver';
+import { promisify } from 'util';
 import { PackageManager } from '../lib/config/workspace-schema';
 import { getConfiguredPackageManager } from './config';
 
-function supports(name: string): boolean {
+const execPromise = promisify(exec);
+async function supports(name: PackageManager): Promise<boolean> {
   try {
-    execSync(`${name} --version`, { stdio: 'ignore' });
+    await execPromise(`${name} --version`);
```

Comment:
Nit: Would it be better to look to see if the program is on the path rather than executing it? It's hard to predict what the program will do and could be unnecessarily slow or lead to side effects (ie. updating the `package-lock.json`, prompting a login). We also don't know that this is actually NPM/Yarn/pnpm, it could be anything the user has configured with that name.

I don't know if `isInPathEnv('npm')` exists, but might be worth checking if that can be a better alternative. If not, this is fine as is, I admit this concern is pretty edge-casey.

Reply:
You definitly made some good points here, although I don't think there is a Node.js similar to `isInPathEnv`.